x-pack/legacy/plugins/lens/public/editor_frame_plugin/editor_frame/state_management.ts

```diff
@@ -64,6 +64,21 @@ export type Action =
       newDatasourceId: string;
     };
 
+function getActiveDatasourceIdFromDoc(doc?: Document) {
+  if (!doc) {
+    return null;
+  }
+
+  const [initialDatasourceId] = Object.keys(doc.state.datasourceStates);
```

Comment:
Well you were asking about what the role of `activeDatasourceId` is, and this is it- even though we only have one datasource now, in the future we're expecting multiple datasources. This is grabbing the first datasource ID in position order, but that's not always going to be meaningful. So my general preference is to keep `activeDatasourceID` in the mapping for this kind of case

Reply:
I think using the first one in the doc is fine. `activeDatasourceId` is a UI concern and shouldn't be persisted, in my opinion.